File: Web-API/Web-API_With_JWT/Frontend1/Frontend/src/Models/movie.ts
export interface Movie {
  id?: number;
  imdbId: string;
  title: string;
  year: number;
  img: string;           
  poster?: string;       
  actors: string[];
  genre: string;
  plot: string;
  director: string;
  writer: string;
  rated: string;
  runtime: string;
  imdbRating: string;
  rank: number;        
  aka: string;          
  imbd_url: string;      
  imbd_iv?: string;
  width?: number;
  height?: number;
}

export interface WatchlistItem {
  id: number;
  userId: number;
  movieId: number;
  dateAdded: string;
  movie: Movie;
}

export interface WatchedItem {
  id: number;
  userId: number;
  movieId: number;
  timesWatched: number;
  firstWatchedAt: string;
  lastWatchedAt: string;
  movie: Movie;
}

export interface GenreStat {
  genre: string;
  count: number;
}

export interface YearlyStat {
  year: number;
  count: number;
}

export interface SummaryStats {
  totalWatched: number;
  uniqueMovies: number;
  mostWatchedMovie: string;
  mostWatchedCount: number;
  averageImdbRating: number;
}

File: Web-API/Web-API_With_JWT/Models/Movie.cs
using System.ComponentModel.DataAnnotations;
using System.Reflection;

namespace Ass3WebApi.Models
{
    public class Movie
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string ImdbId { get; set; } = string.Empty;

        [Required]
        public string Title { get; set; } = string.Empty;

        public int Year { get; set; } = 0;

        public string Poster { get; set; } = string.Empty;

        public string Actors { get; set; } = string.Empty;

        public string Genre { get; set; } = string.Empty;

        public string Plot { get; set; } = string.Empty;

        public string Director { get; set; } = string.Empty;

        public string Writer { get; set; } = string.Empty;

        public string Rated { get; set; } = string.Empty;

        public string Runtime { get; set; } = string.Empty;

        public string ImdbRating { get; set; } = string.Empty;
    }
}
